### DISCUSSION TITLE
建议下载 HEVC 视频时（可选地）将 tag 设为 hvc1

### DISCUSSION THREAD

**Discussion by qianbinbin** [the1812/Bilibili-Evolved#4135] [功能建议 / Ideas]
hev1 vs hvc1: https://stackoverflow.com/questions/63468587/what-hevc-codec-tag-to-use-with-fmp4-hvc1-or-hev1

默认 HEVC 下载后 tag 为 hev1，这在 macOS 上是不兼容的，缩略图无法显示，自带的播放器也无法打开。第三方的 IINA 可以，但还是无法显示缩略图。

如果能可以增加一个**可选开关**，在下载时自动设置就好了。不过我不清楚前端能否实现这样的功能。

附使用 `ffmpeg` 转换方法（不会重新编码）：

```sh
$ ffmpeg -i test.mp4 -c:v copy -tag:v hvc1 -c:a copy test.hvc1.mp4
```

谢谢。

**Comment by the1812** [comment]
和 dash 格式合并类似, 这个需要 ffmpeg 处理, 前端不好接入

**Comment by drlcl** [comment]
> hev1 vs hvc1: https://stackoverflow.com/questions/63468587/what-hevc-codec-tag-to-use-with-fmp4-hvc1-or-hev1
> 
> 默认 HEVC 下载后 tag 为 hev1，这在 macOS 上是不兼容的，缩略图无法显示，自带的播放器也无法打开。第三方的 IINA 可以，但还是无法显示缩略图。
> 
> 如果能可以增加一个**可选开关**，在下载时自动设置就好了。不过我不清楚前端能否实现这样的功能。
> 
> 附使用 `ffmpeg` 转换方法（不会重新编码）：
> 
> ```shell
> $ ffmpeg -i test.mp4 -c:v copy -tag:v hvc1 -c:a copy test.hvc1.mp4
> ```
> 
> 谢谢。

建议修改一下命令，改成：
`ffmpeg -i input.mp4 -c copy -tag:v hvc1 output.mp4`
